overlay/agent.hpp

```diff
@@ -13,6 +13,13 @@
 #include <overlay/agent_metrics.hpp>
 #include <overlay/messages.hpp>
 
+// `REGISTERING` is used as an enum value, but it's actually defined as a
+// constant in the Windows SDK.
+#ifdef __WINDOWS__
+#undef REGISTERING
+#undef REGISTERED
+#endif // __WINDOWS__
+
```

Comment:
Do I understand correctly that the conflicting definitions just come from this file (ManagerProcess::State below), and are not #include`d from some Mesos headers? 

If so, can you rename them now or, at least,  leave `//TODO(akornatskyy): Rename enum values in ManagerProcess::State in this file to avoid name collistion.` ?

Reply:
added TODO